src/polycubed/src/rest_server.cpp

```diff
@@ -931,5 +937,157 @@ const std::string &RestServer::getPort() {
     return port;
 }
 
+void RestServer::create_metrics() {
+  logger->info("loading metrics from yang files");
+  try {
+    // uri and metricHandler are not necessary, I use pistache to registry metrics
+    registry = std::make_shared<prometheus::Registry>();
+    std::vector<std::string> name_services = core.get_servicectrls_names_vector();
+
+    // loop on all services (not cubes)
+    for (size_t i = 0; i < name_services.size(); i++) {
+      
+      // for every service I get the array of InfoMetric
+      auto metrics_service = core.get_service_controller(name_services[i]).get_infoMetrics();
+      for(auto metric: metrics_service)  {
+        if (metric.typeMetric == "C") {
+          map_metrics[name_services[i]].counters_map.emplace(metric.nameMetric,std::ref(prometheus::BuildCounter()
+                  .Name(metric.nameMetric)
+                  .Help(metric.helpMetric)
+                  .Register(*registry)));
+        } else if (metric.typeMetric == "G") {
+          map_metrics[name_services[i]].gauges_map.emplace(metric.nameMetric, std::ref( prometheus::BuildGauge()
+                  .Name(metric.nameMetric)
+                  .Help(metric.helpMetric)
+                  .Register(*registry)));
+        }
+      }
+    }
+    // all metrics created are put into collectalbes_ thanks to registry
+    // collectables_ can collects al types of metrics (counter, gauge, histogram, summary)
+    collectables_.push_back(registry);
+  } catch (const std::runtime_error &e) {
+    logger->error("{0}", e.what());
+  }
+}
+
+
+// expose OpenMetrics
+void RestServer::get_metrics(const Pistache::Rest::Request &request,
```

Comment:
Add a brief comment here.
It would be nice if you report also an example of the returned values.

Reply:
Added :+1: